### PR TITLE
added missing Role for director search

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/3309

*Description of changes:*

*Checklist:*
I confirm that below checklist items are addressed in this pull request; (check the boxes applicable)
- [ ] No lint errors
- [ ] No test case failures and proper coverage for all modules/classes
- [ ] Updated the deployment configs for new environment variable(s)
- [ ] Updtaed the postman collection in entity repository (https://github.com/bcgov/entity/tree/master/api-e2e/postman) for e2e tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/services/products.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def create_product_subscription(org_id, subscription_data: Tuple[Dict[str, Any]]
                 subscriptions_model_list.append(product_subscription)
             else:
                 raise BusinessException(Error.DATA_NOT_FOUND, None)
-            if subscription.get('product_roles') is not None:
+            if subscription.get('productRoles') is not None:
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/services/products.py:56]: Can we give all roles for that product by default? Else UI will need to query all the roles and then pass it.

Comment (saravanpa-aot) [reply auth-api/src/auth_api/services/products.py:56]: u mean if nothing is passed by default ? The only reason i put it is if we ever put the roles in UI as well..a role selection as well

Comment (saravanpa-aot) [reply auth-api/src/auth_api/services/products.py]: I added the new logic..Is that good ?
If no roles are passed from UI for a product , all roles gets added..
If the logic is okay , will add more tests..lmk

**Comment by saravanpa-aot** [comment]
I will add the tests in the next PR..merging this

**Pull request merged**